### PR TITLE
Use common prototypeMethods from keystone-storage-namefunctions package

### PR DESCRIPTION
**Pull request by JedWatson**
These changes are dependent on a new version of the `keystone-storage-namefunctions` package being published, wanted to push this out for review first.

See the latest commits in master of that repo for the files this references and associated documentation:

https://github.com/keystonejs/keystone-storage-namefunctions

cc/ @andrewlinfoot these would be used for the cloudinaryImage PR #3441

### CODE REVIEW

**Comment by mxstbr** [comment]
Oooh nice idea, like it!

File: lib/storage/adapters/fs/index.js [hunk 1]
```diff
@@ -59,6 +60,13 @@ FSAdapter.SCHEMA_FIELD_DEFAULTS = {
 };
 
 /**
+	Inherit common prototype behaviours for generating and retrying filenames
+	from the keystone-storage-namefunctions package
+*/
+FSAdapter.prototype._getFilename = prototypeMethods.getFilename;
+FSAdapter.prototype._retryFilename = prototypeMethods.retryFilename;
```
Comment (andrewlinfoot) [inline lib/storage/adapters/fs/index.js]: In the `prototypeMethod.getFilename` function it expects a function `this.retryFilename` not `this._retryFilename`. See: https://github.com/keystonejs/keystone-storage-namefunctions/commit/dc3ef5cc0ea632dac7b2f48a6104c21da8033566

**Comment by JedWatson** [comment]
@andrewlinfoot I've just published a new version of `keystone-storage-namefunctions` - I'll update the package.json with this PR and get it merged so you can wrap up #3441

**Comment by JedWatson** [comment]
I've tested this in the keystone test project, it looks good to merge.

**Pull request merged**